src/app.ts

```diff
@@ -267,13 +302,15 @@ async function clearAndReloadSpidStrategy(
   let newSpidStrategy: SpidStrategy | undefined;
   try {
     // Inject a new SPID Strategy generate function inside the container
-    container.register(SPID_STRATEGY, defaultModule.loadSpidStrategy());
-    newSpidStrategy = await container.resolve(SPID_STRATEGY);
+    // tslint:disable-next-line: no-object-mutation
+    defaultModule.SPID_STRATEGY = defaultModule.loadSpidStrategy();
```

Comment:
can we have `defaultModule.SPID_STRATEGY` be the result of the promise? i.e.

```suggestion
    defaultModule.SPID_STRATEGY = await defaultModule.loadSpidStrategy();
```

Reply:
I think that is possible, but is needed a refactoring of app, AuthController and tests. If you want I can make the refactory inside this PR

Reply:
fixed in 7aebb53